ArduPlane/ArduPlane.cpp

```diff
@@ -732,25 +732,43 @@ bool Plane::set_target_location(const Location& target_loc)
 // set target location (for use by scripting)
 bool Plane::get_target_location(Location& target_loc)
 {
-    switch (control_mode->mode_number()) {
-    case Mode::Number::RTL:
-    case Mode::Number::AVOID_ADSB:
-    case Mode::Number::GUIDED:
-    case Mode::Number::AUTO:
-    case Mode::Number::LOITER:
+    target_loc = next_WP_loc;
+    return true;
+}
+
+/*
+  update_target_location() works in all auto navigation modes
+ */
+bool Plane::update_target_location(const Location &old_loc, const Location &new_loc)
+{
+    if (!old_loc.same_latlon_as(next_WP_loc)) {
+        return false;
+    }
+    int32_t alt1, alt2;
+    if (!old_loc.get_alt_cm(Location::AltFrame::ABSOLUTE, alt1) ||
+        !next_WP_loc.get_alt_cm(Location::AltFrame::ABSOLUTE, alt2) ||
+        alt1 != alt2) {
```

Comment:
Could also do this, probably does not make much difference either way. 
```suggestion
    ftype alt_diff;
    if (!old_loc.get_alt_distance(next_WP_loc, alt_diff) ||
        !is_zero(alt_diff)) {
```

Reply:
nicer, thanks. changed